### ISSUE TITLE
Is it recommended to use Reference table to Get IIDs in Batch commands?

### ISSUE THREAD

**Issue by smokyMount** [k2view-academy/K2View-Academy#927]
I tried to run a batch command using common table from fabric interface and it executed the batch correctly but then if I run a query from same session it fails with connection is closed error.

```

fabric>BATCH DUMMY FROM fabric USING ("select ID from COMMON.IID_LIST") FABRIC_COMMAND="sync_instance DUMMY.?";
|Batch id                            |Execution id                        |Execution succeeded|Execution failed|Total|Added|Updated|Unchanged|Duration|
+------------------------------------+------------------------------------+-------------------+----------------+-----+-----+-------+---------+--------+
|852197db-ec56-4a38-8958-3c1fdaef98b7|b27bf4e3-4a27-4b17-ad29-6d56fe23cbf8|10                 |0               |10   |10    |0      |0       |1       |

(1 rows)
fabric>select * from COMMON.IID_LIST;
Connection is already closed

```
Then I tried using a FabricRemote interface and it did not throw any such errors. 
```
fabric>BATCH DUMMY FROM FabricRemote USING ("select IID from COMMON.IID_LIST") FABRIC_COMMAND="sync_instance DUMMY.?";
|Batch id                            |Execution id                        |Execution succeeded|Execution failed|Total|Added|Updated|Unchanged|Duration|
+------------------------------------+------------------------------------+-------------------+----------------+-----+-----+-------+---------+--------+
|edd7612f-d46d-4213-b1d8-299234ff5d4f|c4c2a4de-638c-4f57-9df2-28fcad6f0cc4|10                 |0               |10   |0    |10     |0        |1       |

(1 rows)
fabric>select IID FROM IID_LIST;
|IID|
+------------------+
|1                 |
|2                 |
|3                 |
|4                 |
|5                 |
|6                 |
|7                 |
|8                 |
|9                 |
|10                |

(10 rows)
```

What is the recommendation from product? Can we use fabric interface or do we need to define a FabricRemote interface and use that in Batch commands?

**Comment by lionbro1** [comment]
Are you using Fabric 7.1.x release? as this issue was resolved on latest releases.

**Comment by smokyMount** [comment]
We are on 6.5.9 HF15

**Comment by lionbro1** [comment]
Ok, fix is available on 7.1 release.
